addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -394,111 +365,140 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
-                } else {
-                    channelDiskUsage = diskUsage.getDiskUsage();
+                        } else {
+                            channelDiskUsage = diskUsage.getDiskUsage();
+                        }
+                    }
                 }
+            } else {
+                // Make sure old data is cleared
+                channelCpuLoad = "";
+                channelDiskUsage = "";
+            }
+        } catch (Exception ex) {
+            logger.error("{}: context='refreshThing' tag='exception' Exception thrown when refreshing bridge='{}'",
+                    getLogIdentifier(), getThing().getBridgeUID(), ex);
+        } finally {
+            if (zoneMinderServerProxy != null) {
+                releaseSession();
             }
 
-        } else {
-            _online = false;
-            // Make sure old data is cleared
-            channelCpuLoad = "";
-            channelDiskUsage = "";
         }
 
         /*
          * Update all channels on Bridge
          */
         for (Channel channel : channels) {
-            updateChannel(channel.getUID());
+
+            if (isLinked(channel.getUID().getId())) {
+                updateChannel(channel.getUID());
+            }
         }
 
         /*
          * Request Things attached to Bridge to refresh
          */
         for (Thing thing : things) {
             try {
-
                 if (thing.getThingTypeUID().equals(ZoneMinderConstants.THING_TYPE_THING_ZONEMINDER_MONITOR)) {
-                    Thing thingMonitor = thing;
                     ZoneMinderBaseThingHandler thingHandler = (ZoneMinderBaseThingHandler) thing.getHandler();
 
-                    thingHandler.refreshThing(session, DataRefreshPriorityEnum.SCHEDULED);
+                    if (thingHandler != null) {
+                        thingHandler.refreshThing(curPriority);
+                    }
                 }
-
-            } catch (NullPointerException ex) {
-                // This isn't critical (unless it comes over and over). There seems to be a bug so that a null
-                // pointer exception is coming every now and then.
-                // HAve to find the reason for that. Until thenm, don't Spamm
-                logger.debug("{}: Method 'refreshThing()' for Bridge {} failed for thing='{}' - Exception='{}'",
-                        getLogIdentifier(), this.getZoneMinderId(), thing.getUID(), ex.getMessage());
-
-                // Other exceptions has to be shown as errors
             } catch (Exception ex) {
-                logger.error("{}: Method 'refreshThing()' for Bridge {} failed for thing='{}' - Exception='{}'",
-                        getLogIdentifier(), this.getZoneMinderId(), thing.getUID(), ex.getMessage());
+                logger.error("{}: context='refreshThing' tag='exception' Exception thrown when refreshing thing='{}'",
```

Comment:
In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Error should only be used:

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. 

Please consider to change this and all other error logs to use the correct logging levels and if possible change the ONLINE/OFFLINE status of thing / bridge.